### PR TITLE
RTL fixes

### PR DESCRIPTION
**Pull request by yosilevy**
## Proposed change

Various RTL fixes
Using ha-list-item to fix drop down problems in entity picker
Fix check list item to properly position in RTL due to lack of support (same fix as for ha-button-menu previously done)
Ha-combo-box - fixing vaadin combo box item wrong RTL spacing to match same spacing as LTR
Fix config -> integrations spacing

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-check-list-item.ts [hunk 1]
```diff
@@ -6,6 +6,19 @@ import { customElement } from "lit/decorators";
...
+    super.firstUpdated();
+
+    if (document.dir === "rtl") {
+      this.updateComplete.then(() => {
+        const style = document.createElement("style");
+        style.innerHTML =
+          "span.mdc-deprecated-list-item__graphic { margin-left: var(--mdc-list-item-graphic-margin, 16px) !important; margin-right: 0px !important;}";
+        this.shadowRoot!.appendChild(style);
+      });
+    }
```
Comment (bramkragten) [inline src/components/ha-check-list-item.ts:19]: why not just add this to the `styles` getting, like the override below for `--mdc-theme-secondary`?

File: src/components/ha-combo-box.ts [hunk 2]
```diff
@@ -304,6 +305,17 @@ export class HaComboBox extends LitElement {
     }
   }
 
+  private _handleRTL(overlay: HTMLElement) {
+    overlay
+      .querySelectorAll("vaadin-combo-box-scroller > vaadin-combo-box-item")
+      .forEach((item) => {
+        const elem = item as HTMLElement;
+        if (elem.dir === "rtl") {
+          elem.style.padding = "0px";
```
Comment (bramkragten) [inline src/components/ha-combo-box.ts:314]: Can we add this to https://github.com/home-assistant/frontend/pull/15233/files#diff-32fb6be2475d4e810c29381a10e4c784eb279630c924063afafe53fd9a4d43dfL22-L23 `registerStyles` for "vaadin-combo-box-item"?

Comment (yosilevy) [reply src/components/ha-combo-box.ts:314]: I started with that. Those styles are nowhere to be seen. Seems like it's broken?

Comment (bramkragten) [reply src/components/ha-combo-box.ts:314]: I see them? 
CleanShot 2023-01-30 at 12 47 00

Comment (bramkragten) [reply src/components/ha-combo-box.ts:314]: Also the padding on that element already is set to `0` right? What are we trying to fix exactly?

File: src/components/ha-check-list-item.ts (hunk 1 above)
Comment (yosilevy) [reply src/components/ha-check-list-item.ts:19]: Done

File: src/components/ha-combo-box.ts (hunk 2 above)
Comment (yosilevy) [reply src/components/ha-combo-box.ts:314]: I added !important since it wasn't being applied in RTL. Don't ask me why.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**